inc/question.class.php

```diff
@@ -1186,4 +1186,93 @@ public static function dropdownForForm($formId, $crit, $name, $value) {
       $items = $question->getQuestionsFromFormBySection($formId, $crit);
       Dropdown::showFromArray($name, $items, []);
    }
+
+   /**
+    * Get target subdata (conditions, regexes or ranges)
+    *
+    * @param string $target target table
+    * @param id|array $condition condition
+    * @return array
+    */
+   public static function getQuestionSubdata($target, $condition) {
+      global $DB;
+
+      $validTargets = [
+         \PluginFormcreatorQuestion_Condition::getTable(),
+         \PluginFormcreatorQuestionRegex::getTable(),
+         \PluginFormcreatorQuestionRange::getTable(),
+      ];
+
+      if (array_search($target, $validTargets) === false) {
+         throw new \InvalidArgumentException("Invalid target ('$target')");
```

Comment:
need a try / catch somewhere to avoid an Error 500 if this exception is triggered

Reply:
This kind of exceptions should never be caught (or would need to be caught in a global try catch at the top level of the framework but this doesn't exist in GLPI). This exception should never occur in a correct implementation, It therefore does not make sense to try to catch and handle it. Instead, the code that triggered the exception should be fixed.